bindingcollectionadapter-recyclerview/src/main/java/me/tatarka/bindingcollectionadapter2/BindingRecyclerViewAdapter.java

```diff
@@ -250,6 +250,12 @@ public long getItemId(int position) {
         return itemIds == null ? position : itemIds.getItemId(position, items.get(position));
     }
 
+    private void tryGetLifecycleOwner() {
+        if (lifecycleOwner == null || lifecycleOwner.getLifecycle().getCurrentState() == Lifecycle.State.DESTROYED) {
```

Comment:
The lifecycle state check is here because it is possible that the same adapter is used across configuration changes.   This allows the lifecycle owner to be changed to the new Activity/Fragment after the configuration change.